### ISSUE TITLE
cannot get foreground and background rois in ./lib/layer_utils/proposal_target_layer.py

### ISSUE THREAD

**Issue by hedes1992** [ruotianluo/pytorch-faster-rcnn#63]
In the line 138-139 of ./lib/layer_utils/proposal_target_layer.py
In my own dataset(car data from kitti-detection, just 2-class, car and background), at the beginning of this end2end training, it meet a small car, and all the overlaps between rois and this gt_bbox is lower than 0.01, so there are no foreground and background rois.
current code just use a pdb, but no post-processing. I guess this batch might be discarded?
Is there any other better method to avoid this? Thank you

But I ran the code on coco and pascal voc, there're no bug like this. I guess the reason is there are multiple classes...

**Comment by ruotianluo** [comment]
If there is a car in the image but no roi match that one, then you should consider changing the anchor size maybe.

**Comment by hedes1992** [comment]
@ruotianluo  Yes, I change the thresh to 0.0, and get a good result on training set. Thanks
![image](https://user-images.githubusercontent.com/14728901/36882039-866b566c-1e0c-11e8-8c2d-043901d8fc8a.png)

**Comment by Saharkakavand** [comment]
@hedes1992 
I trained faster rcnn with kitti dataset, could you let me know how do you evaluate it?

**Comment by hedes1992** [comment]
> @hedes1992
> I trained faster rcnn with kitti dataset, could you let me know how do you evaluate it?

I can only remember this repo can do test on kitti, then use the results to plot the PR curve by official toolkit from kitti[devkir_object.zip](https://s3.eu-central-1.amazonaws.com/avg-kitti/devkit_object.zip), which is from the [2d-object detection webpage](http://www.cvlibs.net/datasets/kitti/eval_object.php?obj_benchmark=2d)

**Comment by Saharkakavand** [comment]
@hedes1992  thank you.
